rep-2014.rst

```diff
@@ -0,0 +1,243 @@
+
+The following diagram shows the proposed methodology for benchmarking performance in ROS 2 which consists of 3 steps:
+
+::
+
+                                                                
+                                                +--------------+
+                    +----------------+  rebuild |              |
+                    |                +---------->              |
+  start  +----------> 1. trace graph |          | 2. benchmark +----------> 3. report
+                    |                |          |              |
+                    +----+------^--^-+          |              |
+                         |      |  |            +-------+------+
+                         |      |  |                    |
+                         +------+  |                    |
+                           LTTng   +--------------------+
+                                       re-instrument
+
+
+1. instrument both the target ROS 2 abstraction/application using `LTTng <https://lttng.org/>`_. Refer to `ros2_tracing <https://github.com/ros2/ros2_tracing>`_ for tools, documentation and ROS 2 core layers tracepoints;
+2. trace and benchmark the ROS 2 application;
+3. create performance reports with the results of the benchmarking.
+
+
+Reference implementation and recommendations
+============================================
+
+The reader is referred to `ros2_tracing <https://github.com/ros2/ros2_tracing>`_ and `LTTng <https://lttng.org/>`_ to familiarize herself with the tools and the methodology of collecting and analyzing performance data. In addition, [3]_ and [4]_ present comprehensive descriptions of the `ros2_tracing <https://github.com/ros2/ros2_tracing>`_ tools and the `LTTng <https://lttng.org/>`_ infrastructure.
+
+Reference implementations complying with the recommendations of this REP can be found in literature for applications like perception and mapping [5]_, hardware acceleration [6]_ [7]_ or self-driving mobility [8]_. A particular example of interest for the reader is the instrumentation of the `image_pipeline <https://github.com/ros-perception/image_pipeline/tree/humble/>`_ ROS 2 package [12]_, which is a set of nodes for processing image data in ROS 2. The `image_pipeline <https://github.com/ros-perception/image_pipeline/tree/humble/>`_ package has been instrumented with LTTng probes available in the ROS 2 `Humble` release, which results in various perception Components (e.g. `RectifyNode <https://github.com/ros-perception/image_pipeline/blob/ros2/image_proc/src/rectify.cpp#L82/>`_ *Component*) leveraging intrumentation which if enabled, can help trace the computational graph information flow of a ROS 2 application using such Component. The results of benchmarking the performance of `image_pipeline <https://github.com/ros-perception/image_pipeline/tree/humble/>`_ are available in [13]_ and launch scripts to both trace and analyze perception graphs available in [14]_.
```

Comment:
@vmayoral as [I mentioned on Discourse](https://discourse.ros.org/t/rep-2014-rfc-benchmarking-performance-in-ros-2/27770/14), we could perhaps add a sentence to mention use-cases like heterogeneous systems and mention that `ros2_tracing` is probably not the _only_ tool that would be used.

Reply:
I'm think we can relax a bit the language in here and point to tracing in general, while mentioning `ros2_tracing` as one example that's already integrated in ROS 2.